src/prefect/tasks/redis/redis_tasks.py

```diff
@@ -0,0 +1,214 @@
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+import redis
+
+
+class RedisSet(Task):
+    """
+    Task for setting a Redis key-value pair.
+
+    Args:
+        - connection (Redis Connection , optional): if not provided, connection will be established using
+            host, port, db, and password parameters
+        - host (str, optional): name of Redis host, defaults to 'localhost'
+        - port (int, optional): Redis port, defaults to 6379
+        - db (int, optional): redis database index, defaults to 0
+        - password (str, optional): Redis password, defaults to None
+        - redis_key (optional): Redis key to be set, can be provided at initialization or runtime
+        - redis_val (optional): Redis val to be set, can be provided at initialization or runtime
+        - ex (int, optional): if provided, sets an expire flag, in seconds, on 'redis_key' set
+        - px (int, optional): if provided, sets an expire flag, in milliseconds, on 'redis_key' set
+        - nx (int, optional): if set to True, set the value at 'redis_key' to 'redis_val' only
+            if it does not exist, defaults to False
+        - xx (int, optional): if set to True, set the value at 'redis_key' to 'redis_val' only
+            if it already exists, defaults to False
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        connection=None,
+        host: str = "localhost",
+        port: int = 6379,
+        db: int = 0,
+        password: str = None,
+        redis_key=None,
+        redis_val=None,
+        ex: int = None,
+        px: int = None,
+        nx: bool = False,
+        xx: bool = False,
+        **kwargs
+    ):
+        ## connect to Redis
+        if connection:
+            self.connection = connection
+        else:
+            self.connection = redis.Redis(
+                host=host, port=port, db=db, password=password
+            )
```

Comment:
See comment above; unfortunately we can't create the connection in the `__init__` method, and instead must do it in the `run` method.  Additionally, note that even inside the `run` method you shouldn't attach it as an attribute to this `Task`.